src/dtgtk/thumbtable.c

```diff
@@ -1475,46 +1513,71 @@ static gboolean _event_button_release(GtkWidget *widget,
                                       GdkEventButton *event,
                                       gpointer user_data)
 {
+  // we select only in LIGHTTABLE, DARKROOM & MAP mode
+  dt_view_type_flags_t cv = dt_view_get_current();
+  if(cv != DT_VIEW_DARKROOM
+     && cv != DT_VIEW_LIGHTTABLE
+     && cv != DT_VIEW_MAP)
+    return FALSE;
+
   dt_set_backthumb_time(0.0);
+  const dt_imgid_t id = dt_control_get_mouse_over_id();
 
   dt_thumbtable_t *table = (dt_thumbtable_t *)user_data;
 
-  if(table->dragging == FALSE)
-  {
-    // on map view consider click release instead of press
-    dt_view_manager_t *vm = darktable.view_manager;
-    dt_view_t *view = vm->current_view;
-    const dt_imgid_t id = dt_control_get_mouse_over_id();
-    if(dt_is_valid_imgid(id)
-       && event->button == 1
-       && table->mode == DT_THUMBTABLE_MODE_FILMSTRIP
-       && event->type == GDK_BUTTON_RELEASE
-       && !strcmp(view->module_name, "map")
-       && dt_modifier_is(event->state, 0))
-    {
-      DT_DEBUG_CONTROL_SIGNAL_RAISE(darktable.signals,
-                                    DT_SIGNAL_VIEWMANAGER_THUMBTABLE_ACTIVATE, id);
-      return TRUE;
+  if(dt_is_valid_imgid(id)
+     && event->button == 1
+     && event->type == GDK_BUTTON_RELEASE)
+  {
+    if(dt_modifier_is(event->state, GDK_CONTROL_MASK)
+       || dt_modifier_is(event->state, GDK_MOD2_MASK)) // CMD key on macOS
+    {
+      dt_selection_toggle(darktable.selection, id);
     }
-    else if(dt_is_valid_imgid(id)
-            && event->button == 1
-            && table->mode == DT_THUMBTABLE_MODE_FILMSTRIP
-            && event->type == GDK_BUTTON_RELEASE
-            && strcmp(view->module_name, "map")
-            && dt_modifier_is(event->state, 0))
+    else if(dt_modifier_is(event->state, GDK_SHIFT_MASK))
+    {
+      dt_selection_select_range(darktable.selection, id);
+    }
+    else
+    {
+      if(table->mode == DT_THUMBTABLE_MODE_FILMSTRIP
+         && cv == DT_VIEW_DARKROOM)
+      {
+        // if there is more than one selected image then we have at least
+        // one picture selected not counting the currently edited one.
+        // delay the single selection to ensure that if we double-click we
+        // do not unselect all the pictures.
+        if(table->sel_single_cb == 0)
+        {
+          // button released event must be skip
+          if(table->to_selid == -1)
+          {
+            table->to_selid = NO_IMGID;
+          }
+          else
+          {
+            table->to_selid = id;
+            table->sel_single_cb = g_timeout_add(300, _do_select_single, table);
```

Comment:
300ms is rather fast and may be difficult for some users.  Gtk default for double-click is 400.  Even better  would be reading the current setting and using that.